rosbag2_cpp/src/rosbag2_cpp/typesupport_helpers.cpp

```diff
@@ -15,95 +15,32 @@
 #include "rosbag2_cpp/typesupport_helpers.hpp"
 
 #include <memory>
-#include <sstream>
-#include <stdexcept>
 #include <string>
 #include <tuple>
 #include <utility>
 
-#include "ament_index_cpp/get_resources.hpp"
-#include "ament_index_cpp/get_package_prefix.hpp"
-
+#include "rclcpp/typesupport_helpers.hpp"
 #include "rcpputils/shared_library.hpp"
 
-#include "rosidl_runtime_cpp/message_type_support_decl.hpp"
-
 namespace rosbag2_cpp
 {
 
 std::string get_typesupport_library_path(
   const std::string & package_name, const std::string & typesupport_identifier)
 {
-  const char * filename_prefix;
-  const char * filename_extension;
-  const char * dynamic_library_folder;
-#ifdef _WIN32
-  filename_prefix = "";
-  filename_extension = ".dll";
-  dynamic_library_folder = "/bin/";
-#elif __APPLE__
-  filename_prefix = "lib";
-  filename_extension = ".dylib";
-  dynamic_library_folder = "/lib/";
-#else
-  filename_prefix = "lib";
-  filename_extension = ".so";
-  dynamic_library_folder = "/lib/";
-#endif
-
-  std::string package_prefix;
-  try {
-    package_prefix = ament_index_cpp::get_package_prefix(package_name);
-  } catch (ament_index_cpp::PackageNotFoundError & e) {
-    throw std::runtime_error(e.what());
-  }
-
-  auto library_path = package_prefix + dynamic_library_folder + filename_prefix +
-    package_name + "__" + typesupport_identifier + filename_extension;
-  return library_path;
-}
-
-const std::pair<std::string, std::string> extract_type_and_package(const std::string & full_type)
-{
-  std::string package_name;
-  std::string type_name;
-
-  std::tie(package_name, std::ignore, type_name) = extract_type_identifier(full_type);
-
-  return {package_name, type_name};
+  return rclcpp::get_typesupport_library_path(package_name, typesupport_identifier);
```

Comment:
What about changing tests to call `rclcpp` directly?

Reply:
@christophebedard I didn't expect so fast review for the draft PR. :)

I force pushed more changes by removing duplicates complitly. However, some required API are not public in the rclcpp.
This PR will be dependent from another PR in the RCLCPP to make missing API publically visible.